Fix:
tkevent.c:682:3: error: implicit declaration of function 'TkWmProtocolEventProc' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                TkWmProtocolEventProc(winPtr, eventPtr);
                ^
tkevent.c:1447:5: error: implicit declaration of function 'Tcl_CheckForSignal' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
if (Tcl_CheckForSignal (tk_mainInterp, TCL_OK) == TCL_ERROR)
    ^
tkevent.c:1512:5: error: implicit declaration of function 'Tcl_CheckForSignal' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
if (Tcl_CheckForSignal (tk_mainInterp, TCL_OK) == TCL_ERROR)
    ^
--- src/tk/tkevent.c.orig	2018-06-29 04:19:31.000000000 -0500
+++ src/tk/tkevent.c	2022-05-16 21:32:25.000000000 -0500
@@ -21,6 +21,8 @@
 
 #include "tkconfig.h"
 #include "tkint.h"
+#include "tclxtend.h"
+#include "tkwm.h"
 #include <errno.h>
 #include <signal.h>
 #include <sys/time.h>
